openedx/core/djangoapps/user_api/preferences/api.py

```diff
@@ -353,11 +350,15 @@ def validate_user_preference_serializer(serializer, preference_key, preference_v
         PreferenceValidationError: the supplied key and/or value for a user preference are invalid.
     """
     if preference_value is None or unicode(preference_value).strip() == '':
-        message = _(u"Preference '{preference_key}' cannot be set to an empty value.").format(
-            preference_key=preference_key
-        )
         raise PreferenceValidationError({
-            preference_key: {"developer_message": message, "user_message": message}
+            preference_key: {
+                "developer_message": u"Preference '{preference_key}' cannot be set to an empty value.".format(
```

Comment:
Note that you can use noop to avoid having to duplicate the string.

(http://blog.bessas.me/post/65775299341/using-gettext-in-django